src/v2/stack_aggregator.py

```diff
@@ -351,13 +358,168 @@ def execute(request: Dict, persist=True):
+        packages = self._normalized_packages.all_deps_without_pseudo
+        data = self._get_data_from_graph(
+            packages, get_package_details_with_vul_query, '_get_pkg_details_with_vuls')
+        return data['result']['data']
+
+    def get_package_details_from_graph(self) -> Dict[Package, PackageDetails]:
+        """Get dependency data from graph."""
+        graph_response = self._get_package_details_with_vulnerabilities()
+        package_details: List[Tuple[Package, PackageDetails]] = []
+        for pkg in graph_response:
+            package_details.append(self._get_package_details(pkg))
+
+        psedo_pkgs_data = self._get_package_details_from_graph_for_pseudo_versions()
+        for pseudo_pkg in psedo_pkgs_data:
+            pseudo_pkg_details = self._get_golang_package_details(pseudo_pkg)
+            package_details.append(pseudo_pkg_details)
+
+        # covert list of (pkg, package_details) into map
+        return dict(package_details)
+
+    def _get_data_from_graph(self, packages, query, caller=None) -> Dict:
+        """Get package data from graph along with vulnerability."""
+        logger.info('Executing _get_data_from_db.')
+        time_start = time.time()
+        pkgs_with_vuln = {
+            "result": {
+                "data": []
+            }
+        }
+        # get rid of leading white spaces
+        query = inspect.cleandoc(query)
+        bindings = {
+            'ecosystem': self._normalized_packages.ecosystem,
+            'packages': []
+        }
+        # call gremlin in batches of GREMLIN_QUERY_SIZE
+        for packages in _get_packages_in_batch(packages, GREMLIN_QUERY_SIZE):
+            bindings['packages'] = list(packages)
+            started_at = time.time()
+            result = post_gremlin(query, bindings)
+            logger.info(
+                '%s took %0.2f secs for post_gremlin() batch request',
+                self._request.external_request_id, time.time() - started_at)
+            if result:
+                pkgs_with_vuln['result']['data'] += result['result']['data']
+
+        logger.info('%s took %0.2f secs for %s'
+                    'for total_results %d', self._request.external_request_id,
+                    time.time() - time_start, caller, len(pkgs_with_vuln['result']['data']))
+        return pkgs_with_vuln
+
+    def _filter_vulnerable_packages(self, vulnerabilities: List) -> Dict:
+        """Filter out vulnerabilities whose commit sha is out of vuln_commit_rules."""
+        logger.info('Executing filter_vulnerable_packages')
+
+        filter_vulnerabilities = defaultdict(list)
+        gh = GithubUtils()
+        for vuln in vulnerabilities:
+            package_name = vuln.get('package_name', [None])[0]
+            vuln_rules = vuln.get('vuln_commit_date_rules', [None])[0]
+            pseudo_version = self._normalized_packages.version_map.get(package_name)
+            if not pseudo_version:
+                logger.info("Not a Pseudo Version.")
```

Comment:
nit: This should be a debug print, along with package and version details.

Reply:
Ack!